schemasheets/generate_populate.py

```diff
@@ -0,0 +1,343 @@
+import yaml
+from click import command, option, Choice
+from jsonasobj2 import as_dict
+from linkml_runtime import SchemaView
+from schemasheets.conf.configschema import ColumnSettings
+from schemasheets.schema_exporter import SchemaExporter
+from schemasheets.schemasheet_datamodel import TableConfig, ColumnConfig
+import logging
+import pkg_resources
+
+# todo write tests
+
+# todo: include read only slots?
+
+# todo: check metamodel if element is deprecated
+#  (if it can be included in the report, the deprecation status will be reported!)
+#   like subclass_of, subproperty_of ?
+
+# todo check for inlining in addition to checking if a range class has a identifier slot?
+
+# todo support generating an exhaustive and concise report in the same run
+
+# todo: support slots which might require multiple columns for multiple inner keys,
+#  like examples (values and description)
+
+root_classes = ['slot_definition', "class_definition"]  # hard coding the intention to do a class slot usage report
+
+# todo could these be semantically derived from the root_classes? other than lexical string splitting?
+boilerplate_cols = ['slot', 'class']
+
+# these a slots whose ranges are classes with identifier slots, but they still can't be included in the report
+blacklist = [
+    'attributes',
+    'instantiates',
+    'name',
+    'slot_usage',
+    'slots',
+]
+
+# this salvages these slots (by name) from untemplateables
+requires_column_settings = {
+    "examples values": {
+        "name": "examples",
+        "ikm_slot": "value",  # todo: what about description?
+        "ikm_class": "example",
+        "internal_separator": "|",
+
+    },
+    "structured_pattern": {
+        "name": "structured_pattern",
+        "ikm_slot": "syntax",  # todo what about interpolated and partial_match
+        "ikm_class": "pattern_expression",
+        "internal_separator": "|",
+
+    },
+    "unit symbol": {
+        "name": "unit",
+        "ikm_slot": "symbol",
+        # todo what about symbol, exact mappings, ucum_code, derivation, has_quantity_kind, iec61360code
+        "ikm_class": "UnitOfMeasure",
+        "internal_separator": "|",
+    },
+}
+
+debug_report = {
+    "blacklist_skipped": []
+}
+
+
+def setup_logging(log_file=None, log_level=logging.INFO):
+    logging.basicConfig(
+        level=log_level,
+        format="%(asctime)s - %(name)s - %(levelname)s - %(message)s",
+        filename=log_file,
+    )
+
+
+def tabulate_unique_values(list_):
+    """
+    tabulate the number of appearances of each unique values in a list
+
+    Args:
```

Comment:
See https://berkeleybop.github.io/best_practice/

* Always provide docstrings
     * ReST (reStructuredText) » numpy-style docstrings or google style » nothing

Reply:
Added or changed docstrings to ReST style